### ISSUE TITLE
Change frontend folder structure

### ISSUE THREAD

**Issue by twhiston** [ansible-semaphore/semaphore#487]
As discussed in #479 we will change the layout of the frontend resources to separate resources and final compiled assets. The agreed folder structure will be:
```
root
- api
- cli
- etc.....
- web
    - resources
       - pug
       - less
       - unbundled js etc....
    - public
       - html
       - css
       - js
```
In addition to this we should take the opportunity to add a proper package.json, so that we can install everything directly (as we use dep in the frontend), see https://docs.npmjs.com/files/package.json for details. It should include all [dependencies](https://docs.npmjs.com/files/package.json#dependencies) and [dev dependencies](https://docs.npmjs.com/files/package.json#devdependencies) for compilation etc.... The taskfile should be updated to use npm to install the frontend directly.

**Comment by twhiston** [comment]
@fiftin  let me know if you are up for taking this issue :smile:

**Comment by fiftin** [comment]
Yes, I'll do it :)

**Comment by twhiston** [comment]
I like it! The more we get in to the proper package management world the better. I would pick npm over bower I guess as it has such traction in the market. I really like that with a package.json we can get all our dev tools too etc...
If you want to go in this direction for fe dependency management I'm totally on board.

**Comment by twhiston** [comment]
closed with #489
